Google.Api.Gax.Grpc/Rest/HttpRulePathPattern.cs

```diff
@@ -103,15 +109,30 @@ internal sealed class HttpRulePathPatternSegment
             /// Creates a new path pattern segment from a path segment with the bound field, e.g. `{resource.id}` in `v1/resources/{resource.id}`
             /// </summary>
             /// <param name="propertyPath">A </param>
+            /// <param name="boundFieldPattern"></param>
             /// <param name="descriptor"></param>
             /// <returns></returns>
-            internal static HttpRulePathPatternSegment CreateFromBoundField(string propertyPath, MessageDescriptor descriptor)
+            internal static HttpRulePathPatternSegment CreateFromBoundField(string propertyPath,
+                string boundFieldPattern, MessageDescriptor descriptor)
             {
                 int periodIndex = propertyPath.IndexOf('.');
                 bool singleFieldPath = periodIndex == -1;
                 string fieldName = singleFieldPath  ? propertyPath : propertyPath.Substring(0, periodIndex);
 
-                return new HttpRulePathPatternSegment(fieldName, CreatePropertyAccessor(descriptor, propertyPath));
+                var propertyAccessor = CreatePropertyAccessor(descriptor, propertyPath);
+
+                // Per `google/api/http.proto`, if the pattern has  multiple path segments, such as `"{var=foo/*}"`,
```

Comment:
What about `{var=foo/**}` with a match of `a/b/c`? At that point I'd expect something like `foo/a%2Fb%2Fc`... although I'm not entirely sure whether that's correct.

Reply:
I meant the variable part was a/b/c, so the complete value being "foo/a/b/c".

Reply:
That being said, `{var=**}` is specifically called in `http.proto` as leaving `/` unescaped, so adding this now.

Reply:
If the pattern is `foo/**` and the variable is `foo/a/b/c`, the slashes should remain unescaped. I have added that as a test case as well.

Reply:
Great, thanks. That test is exactly what I was looking for (assuming you've read the spec correctly - I haven't checked it).